### PR TITLE
Modifying "Tax" task to remove Single Responsibility Principle violations

### PR DESCRIPTION
**Pull request by ShahdLubb**

### CODE REVIEW

File: TaxServices.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        {
+            return  FlatRateTaxCalculator.Instance;
+        }
+        public static void ApplyFlatRateTax(Product product)
+        {
+            product.TaxCalculator = getFlatRateTaxCalculator();
+        }
+        public static void ApplyTax(Product product, ITaxCalculator TaxCalculator)
+        {
+            product.TaxCalculator = TaxCalculator;
```
Comment (WadeeSami) [inline TaxServices.cs]: why to do this ?
why should external class be concerned with setting the tax calculator for the product ?

File: FlatRateTaxCalculator.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Price_Calculator_kata
+{
+    public class FlatRateTaxCalculator : ITaxCalculator
+    {
+        private static double _TaxPercentageDefault = 20.0;
```
Comment (WadeeSami) [inline FlatRateTaxCalculator.cs]: I suggest to move this to a separate config file where all constants/configs can be changed in one place

File: TaxServices.cs (hunk 1 above)
Comment (ShahdLubb) [reply TaxServices.cs]: I thought there would be other types of Taxes in the future and it is a good idea to handle all different types of Taxes from the same class, but yeah for the current requirements, there is no need.

File: FlatRateTaxCalculator.cs (hunk 2 above)
Comment (ShahdLubb) [reply FlatRateTaxCalculator.cs]: do you mean a  simple text file or a new class for all configurations?

**Review by WadeeSami — APPROVED** [review]

**Pull request merged**